drivers/modem/modem_cmd_handler.c

```diff
@@ -388,7 +388,10 @@ static int _modem_cmd_send(struct modem_iface *iface,
 	}
 
 	iface->write(iface, buf, strlen(buf));
-	iface->write(iface, "\r", 1);
+
+	if (data->eol != NULL) {
+		iface->write(iface, data->eol, strlen(data->eol));
```

Comment:
What about calculating the length only once and storing the length in `modem_cmd_handler_data` struct. The send function is called all the time and it feels useless to calculate the length every time even if it is not changing.